fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulktools/Main.java

```diff
@@ -0,0 +1,278 @@
+                String resLine = resReader.readLine();
+                lineRed++;
+                if (resLine == null) {
+                    System.out.println("\n Read finished!!!!!");
+                    isMore2Read = false;
+                } else {
+                    System.out.print(".");
+                    bufferStream.write(resLine.getBytes());
+                    bufferStream.write(Constants.NDJSON_LINESEPERATOR);
+                }
+
+                if (bufferStream.size() > Constants.COS_PART_MINIMALSIZE
+                        || (segNum < numberOfFiles - 1 && lineRed == numOfRes4Seg)
+                        || !isMore2Read) {
+                    String segName = cosFile2Break + "_seg" + segNum;
+                    if (uploadId == null) {
+                        uploadId = BulkDataUtils.startPartUpload(cosClient, cosBucketName, segName, true);
+                    }
+
+                    if (bufferStream.size() > 0) {
+                        dataPackTags.add(BulkDataUtils.multiPartUpload(cosClient, cosBucketName, segName,uploadId,
+                                new ByteArrayInputStream(bufferStream.toByteArray()), bufferStream.size(), partNum++));
+                        bufferStream.reset();
+                    }
+
+                    if (segNum < numberOfFiles - 1 && lineRed == numOfRes4Seg
+                            || !isMore2Read) {
+                        BulkDataUtils.finishMultiPartUpload(cosClient, cosBucketName, segName, uploadId, dataPackTags);
+                        System.out.println("Finished writting for " + segName);
+                        lineRed = 0;
+                        segNum++;
+                        uploadId = null;
+                        partNum = 1;
+                        dataPackTags.clear();
+                    }
+                }
+        }
+
+    }
+
+
+    /**
+     * Main entry
+     * @param args
+     */
+    public static void main(String[] args) {
+        Main m = new Main();
+        try {
+            m.parseArgs(args);
+
+            long start = System.nanoTime();
+            int totalNum = 0;
+
+            AmazonS3 cosClient = BulkDataUtils.getCosClient(cosCredentialIbm, cosApiKey, cosSrvinstId, cosEndpintUrl,
+                    cosLocation);
+            if (cosClient == null) {
+                logger.warning("Failed to get CosClient!");
+                System.exit(1);
+            }
+
+            S3Object item = cosClient.getObject(new GetObjectRequest(cosBucketName, cosFile2Break));
+            try (S3ObjectInputStream s3InStream = item.getObjectContent();
+                    BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream))) {
+                   totalNum = m.getFhirResourceNumberFromBufferReader(resReader);
+               } catch (Exception ioe) {
+                   logger.warning("Error proccesing file " + cosFile2Break + " - " + ioe.getMessage());
+                   System.exit(1);
```

Comment:
There are a lot of System.exits here... .throw an exception, and handle one time.

Reply:
make sense

Reply:
done